packages/unmock-core/src/generator.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * Implements the logic for generating a response from a service file
+ */
+
+import {
+  CreateResponse,
+  IMock,
+  IResponseCreatorFactoryInput,
+  ISerializedRequest,
+  IUnmockServiceState,
+  RequestToSpec,
+} from "./interfaces";
+import { httpRequestMatcherFactory } from "./matcher";
+
+// json-schema-faker doesn't have typed definitions?
+// @ts-ignore
+import jsf from "json-schema-faker";
+
+export const responseCreatorFactory = (
+  opts: IResponseCreatorFactoryInput,
+): CreateResponse => {
+  const httpRequestMatcher = httpRequestMatcherFactory(opts.mockGenerator);
+  return (req: ISerializedRequest) => httpRequestMatcher(req);
+};
+
+const getPathSchemaFromSpec = (
+  spec: any,
+  method: string,
+  path: string,
+): any => {
+  // TODO: Smarter path matching (to match path parameters, for example)
+  if (spec.paths === undefined || spec.paths[path] === undefined) {
+    // spec.info.title is mandatory in OAS, we assume `spec` is OAS compliant
+    throw new Error(`Can't find path '${path}' for ${spec.info.title}`);
+  }
+  const pathSchema = spec.paths[path][method];
+  if (
+    pathSchema === undefined ||
+    pathSchema.responses === undefined ||
+    Object.keys(pathSchema.responses).length === 0
+  ) {
+    throw new Error(
+      `Can't find response for '${method} ${path}' for ${spec.info.title}`,
+    );
+  }
+  // We return the entire object so we can resolve references
+  // to request parameters etc as needed
+  return pathSchema;
+};
+
+const templateFromResponse = (response: any): any => {
```

Comment:
Also here even a type alias would make it a bit easier to understand what's going in and out, I think?